ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/classload/NoExitVmSecManager.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.shell.invoker.classload;
+
+import java.security.Permission;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A helper class for class load invoker so that
+ * the system.exit calls will not exit the VM.
+ * TODO: Remove this and fix VM exit bug.
```

Comment:
Rather than a custom sec manager, which is an overkill, I would prefer to fix this at the actual place. Say if we are starting the shell, then we don't have to exit the program. We could include this logic to the backend generation phase.

Reply:
I created an issue for this #28048